packages/xo-web/src/common/render-xo-item.js

```diff
@@ -299,6 +299,56 @@ Vdi.defaultProps = {
 
 // ===================================================================
 
+export const Pif = decorate([
+  connectStore(() => {
+    const getObject = createGetObject()
+    const getNetwork = createGetObject(createSelector(getObject, pif => get(() => pif.$network)))
+
+    // FIXME: props.self ugly workaround to get object as a self user
+    return (state, props) => ({
+      pif: getObject(state, props, props.self),
+      network: getNetwork(state, props),
+    })
+  }),
+  ({ id, showNetwork, pif, network }) => {
+    if (pif === undefined) {
+      return unknowItem(id, 'PIF')
+    }
+
+    const { carrier, device, deviceName, vlan } = pif
+
+    return (
+      <span>
+        <Icon icon='network' color={carrier ? 'text-success' : 'text-danger'} /> {device} ({deviceName}
+        {vlan !== -1 && (
+          <span>
+            {' '}
+            -{' '}
+            {_('keyValue', {
+              key: _('pifVlanLabel'),
+              value: vlan,
+            })}
+          </span>
+        )}
+        {showNetwork && network !== undefined && <span> - {network.name_label}</span>})
```

Comment:
If `vlan === -1`, `deviceName === ''` and `showNetwork === false`, we will show empty parenthesis.
[See PR#7022](https://github.com/vatesfr/xen-orchestra/pull/7022).

Reply:
True, but I was tired of adding conditions for everything :innocent: And it was making the code unreadable. Same thing for the `-`'s, I don't test the existence of the previous elements. I'm not sure what we should do there, make the code unreadable but correct, or make it readable but sometimes render a bit ugly. Or handle the whole parenthesis case separately?

Reply:
IMHO, if good for 90% of use cases and usable for the rest, it's good enough for XO5.

Reply:
So if that ok with you: I'm just going to add back the test for the whole group so that I don't introduce a regression compared to #7022 (and I think the code will still be fairly readable) but I'm not going to test the `-`'s.